include/swift/AST/DiagnosticsSema.def

```diff
@@ -2731,6 +2732,25 @@ ERROR(differentiable_attr_non_protocol_type_constraint_req,none,
 ERROR(differentiable_attr_unsupported_req_kind,none,
       "layout requirement are not supported by @differentiable attribute", ())
 
+// @differentiang
+ERROR(differentiating_attr_expected_result_tuple,none,
+      "@differentiating attribute requires function to return a two-element tuple (value and pullback/differential)", ())
+ERROR(differentiating_attr_expected_result_tuple_value_label,none,
+      "@differentiating attribute requires function to return a two-element tuple (first element must have label 'value:')", ())
+ERROR(differentiating_attr_expected_result_tuple_func_label,none,
+      "@differentiating attribute requires function to return a two-element tuple (second element must have label 'pullback:' or 'differential:')", ())
+ERROR(differentiating_attr_result_value_not_differentiable,none,
+      "@differentiating attribute requires function to return a two-element tuple (first element type %0 must conform to 'Differentiable')", (Type))
+ERROR(differentiating_attr_result_func_invalid_parameter,none,
+      "expected %0 to be a function with a single parameter of type %1", (Identifier, Type))
+ERROR(differentiating_attr_no_diff_parameters,none,
+      "@differentiating attribute requires function to have at least one differentiation parameter", ())
+ERROR(differentiating_attr_unexpected_diff_params_type,none,
+      "unexpected differentiation parameters type; got %0, but expected %1", (Type, Type))
```

Comment:
This actually isn't necessary because `Type` arguments are already printed inside single quotes.
```
swift/test/AutoDiff/differentiating_attr_type_checking.swift:45:2: error: unexpected differentiation parameters type; got '(T, T)', but expected 'T'
@differentiating(generic) // expected-error {{unexpected differentiation parameters type; got '(T, T)', but expected 'T'}}
 ^
```

Reply:
Ok!